runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -1746,6 +1746,39 @@ static
 void genSuperClassTest(TR::Node *node, TR::Register *instanceClassReg, bool instanceClassRegCanBeReclaimed, TR::Register *castClassReg, int32_t castClassDepth,
                                             TR::LabelSymbol *falseLabel, TR_ARM64ScratchRegisterManager *srm, TR::CodeGenerator *cg)
    {
+
+   // When castClass is an unknown at compile time check if castClass is an interface or an array.
+   // If so skip superClassTest.
+   bool dynamicCastClass = castClassDepth == -1;
+   if ( dynamicCastClass )
+      {
+      TR::Register *scratchRegister1 = srm->findOrCreateScratchRegister();
+      TR_ASSERT(node->getSecondChild()->getOpCodeValue() != TR::loadaddr,
+            "genSuperClassTest: castClassDepth == -1 is not supported for a loadaddr castClass");
+
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, scratchRegister1, TR::MemoryReference::createWithDisplacement(cg, castClassReg, offsetof(J9Class, romClass)));
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmw, node, scratchRegister1, TR::MemoryReference::createWithDisplacement(cg, scratchRegister1, offsetof(J9ROMClass, modifiers)));
+
+      // Array an check
+      static_assert(J9AccClassArray == 0x10000, "J9AccClassArray must be 0x10000");
+      static_assert(J9AccInterface == 0x200, "J9AccInterface must be 0x200");
+      // Two cases can reach this code, an instanceOf node or a call to Class.cast() converted to a checkCast node
+      if (node->->getOpCodeValue() == TR::instanceof)
+         {
+         generateTestBitBranchInstruction(cg, TR::InstOpCode::tbnz, node, scratchRegister1, 16, falseLabel);
+         generateTestBitBranchInstruction(cg, TR::InstOpCode::tbnz, node, scratchRegister1, 9, falseLabel);
+         }
+      else
+         {
+         generateTestImmInstruction(cg, node, scratchRegister1, 0x400); // 0x400 is immr:imms for 0x10000
+         generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, falseLabel, TR::CC_NE);
+         generateTestImmInstruction(cg, node, scratchRegister1, 0xDEADBEEF); // 0xXXXX is immr:imms for 0x200
```

Comment:
Please use 0x5c0 as `immr:imms` for `tst w0, #0x200`.

Reply:
Great, thank you